includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -503,6 +519,11 @@ public function payment_fields() {
 	 * @return array|null An array with result of payment and redirect URL, or nothing.
 	 */
 	public function process_payment( $order_id, $retry = true, $force_save_source = false, $previous_error = false, $use_order_source = false ) {
+		// Flag for using a deferred intent. To be removed.
+		if ( ! empty( $_POST['wc-stripe-is-deferred-intent'] ) ) {
```

Comment:
I added this to avoid interfering with other flows, like the block checkout one. And also to keep the old logic around until we're not using it anywhere, and to keep the new logic separated to make the cleanup easier later. Glad to handle this differently, tho.

Reply:
I like it! :)